File: backend/src/main/scala/microservices/blog/api/ArchiveBlogPostPlanner.scala
// ArchiveBlogPostPlanner：博客域归档博客文章入口。

package com.typesafe.travel.blog.domain

import cats.effect.IO
import com.typesafe.travel.api.routes.ConnectionApiPlan
import com.typesafe.travel.persistence.blog.BlogPlannerPlainSql

import java.sql.Connection
import java.time.Instant

object ArchiveBlogPostPlanner extends ConnectionApiPlan[BlogPostByIdPlannerRequest, BlogPostResponse]:
  override val name: String = "ArchiveBlogPostPlanner"
  override def plan(input: BlogPostByIdPlannerRequest, connection: Connection): IO[BlogPostResponse] =
    val userId = input.userId.getOrElse(throw new IllegalArgumentException("userId is required"))
    BlogPlannerPlainSql.findPostAuthor(connection, input.postId).flatMap { authorUserId =>
      if authorUserId != userId then IO.raiseError(new IllegalArgumentException("Only the author can archive this post"))
      else
        BlogPlannerPlainSql.updatePostStatus(connection, input.postId, BlogPostStatus.Hidden.toString, Instant.now()) *>
          BlogPlannerPlainSql.findPost(connection, input.postId, Some(userId))
    }

File: frontend/src/microservices/blog/api/ArchiveBlogPostPlanner.ts
// 本文件定义 ArchiveBlogPostPlanner，负责博客域对应接口入口。

import type { BlogPostResponse } from '@/microservices/blog/objects/BlogPostResponse'
import { executeJsonApiRequest } from '@/shared-kernel/api/ApiTransport'

export const archiveBlogPost = (postId: string, payload: { userId: string }): Promise<BlogPostResponse> =>
  executeJsonApiRequest('/ArchiveBlogPostPlanner', 'POST', { postId, userId: payload.userId })
